Devel

ycasg opened the pull request:
Remove assertion from SSFM.py

DanHickstein commented:
Oh, do you just want to get rid of this? I think that I figured out that math.fsum could be used instead of np.sum, to provide more numerical accuracy and avoid the problems. But, if you think that we should get rid of it, I suppose my PR is moot...

ycasg commented:
The entire assertion thing is a relic of me figuring out why FFTW didn't work. If you look carefully, you see that all the assertion is doing is making sure that fftshit isn't changing the magnitude of the array. All that the fftshift function does is rearrange the order of the array, so there is really no point at all to keeping it around. 

At the time that I added this, FFTW was running away to infinity. I thought that perhaps there was a glitch in an fft-related function, so I added lots of checks. Until now they didn't really cause issues, but since they are I think that it's time to clean up.

DanHickstein commented:
Okay, makes sense. I'll close the other PR.

Currently, I think that FFTW is actually disabled for the SSFM via the `USE_PYFFTW = False` line in `FourWaveMixing/global_variables.py`. If I change this to `True`, then I get an error (I do not have PyFFTW installed on this laptop).

As I recall, when I did get PyFFTW working on my other computer (which was quite a journey on the Mac), then the SSFM didn't work any faster. But, perhaps I was not doing something correctly. 

Probably we should move this discussion back to the [FFTW Issue](https://github.com/pyNLO/PyNLO/issues/1), but I'm curious what speedup we should expect from PyFFTW for the array sizes that we are using and if it's really worth the trouble. It certainly makes the installation seem more intimidating.
